src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -2490,6 +2490,20 @@ class DrawSketchHandlerDimension : public DrawSketchHandler
 
     void createVerticalConstrain(int GeoId1, Sketcher::PointPos PosId1, int GeoId2, Sketcher::PointPos PosId2) {
         if (selLine.size() == 1) {
+            // If the line is horizontal (should be without constraint if we're here), then we need to modify
+            // its point or we'll get a null line.
+            const Part::Geometry* geo = Obj->getGeometry(GeoId1);
+            if (!(geo->is<Part::GeomLineSegment>())) {
+                return;
+            }
+            auto* line = static_cast<const Part::GeomLineSegment*>(geo);
```

Comment:
This can be simplified probably, something like
```suggestion
            auto line = Base::freecad_dynamic_cast<Part::GeomLineSegment>(Obj->getGeometry(GeoId1));
            if (!line) {
                return;
            }
```

You could also add helper template to getGeometry like I did in https://github.com/FreeCAD/FreeCAD/pull/17528/commits/8f4ff1827f81a6fb448e7a4415c16057ff3271fe. Then you could skip the dynamic cast part and simply do `Obj->getGeometry<Part::GeomLineSegment>(GeoId1)`